cmd/kube-controller-manager/app/options/options.go

```diff
@@ -432,7 +432,10 @@ func (s KubeControllerManagerOptions) Config(allControllers []string, disabledBy
 		return nil, err
 	}
 
-	leaderElectionClient := clientset.NewForConfigOrDie(restclient.AddUserAgent(kubeconfig, "leader-election"))
+	// shallow copy, do not modify the kubeconfig.Timeout.
+	config := *kubeconfig
+	config.Timeout = s.GenericComponent.LeaderElection.RenewDeadline.Duration
```

Comment:
Would it be sufficient to set this to RetryPeriod? I think then we can get away without the other timeout.

Reply:
No, this timeout is to set for http client, used to prevent blocking forever.

Reply:
I understand what it is used for. If we retry every RetryPeriod anyway, why would we want a client to block for longer than RetryPeriod?

Reply:
In general, the timeout less than `RetryPeriod` can work well. But if the network is unstable, and it takes longer to finish the renew. During this period, this client loses leadership.

Reply:
This is my own thought. Would not mind different voice.